Address safer CPP warnings in PlatformStrategies.h

cdumez opened the pull request:
#### 88a72e722ff1a8d40429c03cee9e35d9eb6216f1
<pre>
Address safer CPP warnings in PlatformStrategies.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=298859">https://bugs.webkit.org/show_bug.cgi?id=298859</a>

Reviewed by Ryosuke Niwa.

* Source/WTF/wtf/CheckedRef.h:
(WTF::downcast):
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::enabledForContext):
(WebCore::MediaSource::canConstructInDedicatedWorker):
* Source/WebCore/Modules/mediastream/ImageCapture.cpp:
(WebCore::createImageBitmap):
* Source/WebCore/Modules/push-api/PushStrategy.h:
* Source/WebCore/Modules/webaudio/DefaultAudioDestinationNode.cpp:
(WebCore::DefaultAudioDestinationNode::createDestination):
* Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::SerializedScriptValue::writeBlobsToDiskForIndexedDB):
* Source/WebCore/fileapi/ThreadableBlobRegistry.cpp:
(WebCore::ThreadableBlobRegistry::registerInternalFileBlobURL):
(WebCore::ThreadableBlobRegistry::registerInternalBlobURL):
(WebCore::ThreadableBlobRegistry::registerBlobURL):
(WebCore::ThreadableBlobRegistry::registerInternalBlobURLOptionallyFileBacked):
(WebCore::ThreadableBlobRegistry::registerInternalBlobURLForSlice):
(WebCore::ThreadableBlobRegistry::blobType):
(WebCore::ThreadableBlobRegistry::blobSize):
(WebCore::ThreadableBlobRegistry::unregisterBlobURL):
(WebCore::ThreadableBlobRegistry::registerBlobURLHandle):
(WebCore::ThreadableBlobRegistry::unregisterBlobURLHandle):
* Source/WebCore/loader/LoaderStrategy.cpp:
* Source/WebCore/loader/LoaderStrategy.h:
* Source/WebCore/page/DeprecatedGlobalSettings.cpp:
(WebCore::DeprecatedGlobalSettings::setAVFoundationEnabled):
(WebCore::DeprecatedGlobalSettings::setGStreamerEnabled):
* Source/WebCore/platform/MediaStrategy.cpp:
* Source/WebCore/platform/MediaStrategy.h:
* Source/WebCore/platform/Pasteboard.cpp:
(WebCore::Pasteboard::allPasteboardItemInfo const):
(WebCore::Pasteboard::pasteboardItemInfo const):
(WebCore::Pasteboard::readString):
(WebCore::Pasteboard::readBuffer):
(WebCore::Pasteboard::readURL):
* Source/WebCore/platform/PasteboardStrategy.h:
* Source/WebCore/platform/PlatformStrategies.cpp:
(WebCore::PlatformStrategies::loaderStrategy):
(WebCore::PlatformStrategies::pasteboardStrategy):
(WebCore::PlatformStrategies::mediaStrategy):
(WebCore::PlatformStrategies::blobRegistry):
(WebCore::PlatformStrategies::pushStrategy):
* Source/WebCore/platform/PlatformStrategies.h:
(WebCore::PlatformStrategies::loaderStrategy): Deleted.
(WebCore::PlatformStrategies::pasteboardStrategy): Deleted.
(WebCore::PlatformStrategies::mediaStrategy): Deleted.
(WebCore::PlatformStrategies::blobRegistry): Deleted.
(WebCore::PlatformStrategies::pushStrategy): Deleted.
(WebCore::PlatformStrategies::~PlatformStrategies): Deleted.
(): Deleted.
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::MediaSessionManagerCocoa):
* Source/WebCore/platform/audio/glib/MediaSessionManagerGLib.cpp:
(WebCore::MediaSessionManagerGLib::MediaSessionManagerGLib):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::WTF_REQUIRES_LOCK):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::Pasteboard::read):
(WebCore::Pasteboard::readPlatformValuesAsStrings):
(WebCore::Pasteboard::readFilePaths):
* Source/WebCore/platform/network/BlobRegistry.cpp:
(WebCore::blobRegistry):
* Source/WebCore/platform/network/BlobRegistry.h:
* Source/WebCore/platform/network/BlobRegistryImpl.cpp:
(WebCore::createBlobResourceHandle):
(WebCore::loadBlobResourceSynchronously):
* Source/WebCore/platform/network/FormData.cpp:
(WebCore::FormData::resolveBlobReferences):
* Source/WebCore/platform/network/cf/FormDataStreamCFNet.cpp:
(WebCore::createHTTPBodyCFReadStream):
* Source/WebCore/platform/network/curl/CurlFormDataStream.cpp:
(WebCore::CurlFormDataStream::CurlFormDataStream):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::initializeMockMediaSource):
* Source/WebKit/WebProcess/FileAPI/BlobRegistryProxy.cpp:
* Source/WebKit/WebProcess/FileAPI/BlobRegistryProxy.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::load):
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.cpp:
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.h:
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.h:
* Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebPlatformStrategies.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebPlatformStrategies.mm:
(WebPlatformStrategies::initializeIfNecessary):

Canonical link: <a href="https://commits.webkit.org/300024@main">https://commits.webkit.org/300024@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/807dd1f12228e1c796055fed820d5420fcf1a3fa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121069 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40765 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31423 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127488 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73150 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2d205d72-b3e5-4270-92ff-74402545f680) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122945 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41467 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49344 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91962 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61177 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6628580b-312c-4cd5-b47c-bc7f9286141b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124021 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33112 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108528 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72646 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/05bc26c9-0960-4ff2-b0f8-8db1d8743c15) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32136 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26631 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71079 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/113199 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/102618 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26810 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130341 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/119589 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47996 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36475 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100572 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48364 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104695 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100474 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25472 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45876 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23928 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44684 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47854 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53567 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/149748 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47325 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37985 "Found 1 new JSC stress test failure: stress/dont-link-virtual-calls-on-compiler-thread.js.no-llint (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50672 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49009 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7ec2f0c15335ee95f8de93d571e61ce15e0835bd. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50739

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ad9682a34eb215c4dcce66ba56e78116ca68122f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50739

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4071911fa5e5d3449ffc602d6bad8fb9723cfb48. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50739

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0cdd1bf9fcac54dac06570cad3c255c242149ff5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50739

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/807dd1f12228e1c796055fed820d5420fcf1a3fa. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50739

cdumez committed:
Address safer CPP warnings in PlatformStrategies.h
https://bugs.webkit.org/show_bug.cgi?id=298859

Reviewed by Ryosuke Niwa.

* Source/WTF/wtf/CheckedRef.h:
(WTF::downcast):
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::enabledForContext):
(WebCore::MediaSource::canConstructInDedicatedWorker):
* Source/WebCore/Modules/mediastream/ImageCapture.cpp:
(WebCore::createImageBitmap):
* Source/WebCore/Modules/push-api/PushStrategy.h:
* Source/WebCore/Modules/webaudio/DefaultAudioDestinationNode.cpp:
(WebCore::DefaultAudioDestinationNode::createDestination):
* Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::SerializedScriptValue::writeBlobsToDiskForIndexedDB):
* Source/WebCore/fileapi/ThreadableBlobRegistry.cpp:
(WebCore::ThreadableBlobRegistry::registerInternalFileBlobURL):
(WebCore::ThreadableBlobRegistry::registerInternalBlobURL):
(WebCore::ThreadableBlobRegistry::registerBlobURL):
(WebCore::ThreadableBlobRegistry::registerInternalBlobURLOptionallyFileBacked):
(WebCore::ThreadableBlobRegistry::registerInternalBlobURLForSlice):
(WebCore::ThreadableBlobRegistry::blobType):
(WebCore::ThreadableBlobRegistry::blobSize):
(WebCore::ThreadableBlobRegistry::unregisterBlobURL):
(WebCore::ThreadableBlobRegistry::registerBlobURLHandle):
(WebCore::ThreadableBlobRegistry::unregisterBlobURLHandle):
* Source/WebCore/loader/LoaderStrategy.cpp:
* Source/WebCore/loader/LoaderStrategy.h:
* Source/WebCore/page/DeprecatedGlobalSettings.cpp:
(WebCore::DeprecatedGlobalSettings::setAVFoundationEnabled):
(WebCore::DeprecatedGlobalSettings::setGStreamerEnabled):
* Source/WebCore/platform/MediaStrategy.cpp:
* Source/WebCore/platform/MediaStrategy.h:
* Source/WebCore/platform/Pasteboard.cpp:
(WebCore::Pasteboard::allPasteboardItemInfo const):
(WebCore::Pasteboard::pasteboardItemInfo const):
(WebCore::Pasteboard::readString):
(WebCore::Pasteboard::readBuffer):
(WebCore::Pasteboard::readURL):
* Source/WebCore/platform/PasteboardStrategy.h:
* Source/WebCore/platform/PlatformStrategies.cpp:
(WebCore::PlatformStrategies::loaderStrategy):
(WebCore::PlatformStrategies::pasteboardStrategy):
(WebCore::PlatformStrategies::mediaStrategy):
(WebCore::PlatformStrategies::blobRegistry):
(WebCore::PlatformStrategies::pushStrategy):
* Source/WebCore/platform/PlatformStrategies.h:
(WebCore::PlatformStrategies::loaderStrategy): Deleted.
(WebCore::PlatformStrategies::pasteboardStrategy): Deleted.
(WebCore::PlatformStrategies::mediaStrategy): Deleted.
(WebCore::PlatformStrategies::blobRegistry): Deleted.
(WebCore::PlatformStrategies::pushStrategy): Deleted.
(WebCore::PlatformStrategies::~PlatformStrategies): Deleted.
(): Deleted.
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::MediaSessionManagerCocoa):
* Source/WebCore/platform/audio/glib/MediaSessionManagerGLib.cpp:
(WebCore::MediaSessionManagerGLib::MediaSessionManagerGLib):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::WTF_REQUIRES_LOCK):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::Pasteboard::read):
(WebCore::Pasteboard::readPlatformValuesAsStrings):
(WebCore::Pasteboard::readFilePaths):
* Source/WebCore/platform/network/BlobRegistry.cpp:
(WebCore::blobRegistry):
* Source/WebCore/platform/network/BlobRegistry.h:
* Source/WebCore/platform/network/BlobRegistryImpl.cpp:
(WebCore::createBlobResourceHandle):
(WebCore::loadBlobResourceSynchronously):
* Source/WebCore/platform/network/FormData.cpp:
(WebCore::FormData::resolveBlobReferences):
* Source/WebCore/platform/network/cf/FormDataStreamCFNet.cpp:
(WebCore::createHTTPBodyCFReadStream):
* Source/WebCore/platform/network/curl/CurlFormDataStream.cpp:
(WebCore::CurlFormDataStream::CurlFormDataStream):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::initializeMockMediaSource):
* Source/WebKit/WebProcess/FileAPI/BlobRegistryProxy.cpp:
* Source/WebKit/WebProcess/FileAPI/BlobRegistryProxy.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::load):
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.cpp:
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.h:
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.h:
* Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebPlatformStrategies.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebPlatformStrategies.mm:
(WebPlatformStrategies::initializeIfNecessary):

Canonical link: https://commits.webkit.org/300024@main
